starter-code/app.js

```diff
@@ -1,5 +1,73 @@
 const express = require('express');
+var bodyParser = require("body-parser");
 const app = express();
-const Chuck  = require('chucknorris-io');
+const Chuck = require('chucknorris-io');
 const client = new Chuck();
+const ejs = require('ejs')
 
+//Here we are configuring express to use body-parser as middle-ware.
+app.use(bodyParser.urlencoded({ extended: true }));
+
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+
+app.get('/', (req, res, next) => {
+  res.render('index')
+});
+
+app.get('/random', (req, res, next) => {
+  let data = {
+    test: 'woo'
+  }
+
+  client.getRandomJoke().then((response) => {
+    data['joke'] = response.value
+    res.render('random', data)
+  }).catch((error) => {
+    throw error
+  })
+});
+
+app.get('/categories', (req, res, next) => {
+  let data = {
+    test: 'woo'
+  }
+
+  client.getJokeCategories().then((response) => {
+    data['categories'] = response
+    console.log('***** DATA *****', data)
+    res.render('categories', data)
+  }).catch((error) => {
+    throw error
+  })
+});
+
+app.get('/category', (req, res, next) => {
```

Comment:
You're supposed to use the `/categories` route.

Reply:
We thought we got them all :disappointed: